File: codeforces/cf1536A_omkar_and_bad_story.py
# uninhm
# https://codeforces.com/contest/1536/problem/A
# greedy

t = int(input())

solution = ' '.join(map(str, range(101)))
for _ in range(t):
    n = int(input())
    a = [int(i) for i in input().split()]
    if any(i<0 for i in a):
        print("NO")
        continue
    print("YES")
    print(101)
    print(solution)

File: codeforces/cf1536A_omkar_and_bad_story.hs
-- uninhm
-- https://codeforces.com/contest/1536/problem/A
-- greedy

import Control.Monad (replicateM_)

getInt = read <$> getLine
getInts = map read . words <$> getLine

solution = unlines [ "YES"
                   , "101"
                   , unwords (map show [0..100])
                   ]

main = do
  t <- getInt
  replicateM_ t $ do
    n <- getInt
    a <- getInts
    if any (< 0) a then
      putStrLn "NO"
    else
      putStr solution
